cmd/soroban-cli/src/commands/tx/new/revoke_sponsorship.rs

```diff
@@ -0,0 +1,139 @@
+use clap::Parser;
+use soroban_sdk::xdr;
+
+use super::clawback_claimable_balance::parse_balance_id;
+use crate::{commands::tx, config::address, tx::builder};
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub tx: tx::args::Args,
+
+    #[command(flatten)]
+    pub op: Args,
+}
+
+#[derive(Debug, clap::Args, Clone)]
+pub struct Args {
+    /// Account ID (required for all sponsorship types)
+    #[arg(long)]
+    pub account_id: address::UnresolvedMuxedAccount,
+
+    /// Asset for trustline sponsorship (format: CODE:ISSUER or native)
+    #[arg(long, group = "sponsorship_type")]
+    pub asset: Option<builder::Asset>,
+
+    /// Data name for data entry sponsorship
+    #[arg(long, group = "sponsorship_type")]
+    pub data_name: Option<String>,
+
+    /// Offer ID for offer sponsorship
+    #[arg(long, group = "sponsorship_type")]
+    pub offer_id: Option<u64>,
+
+    /// Pool ID for liquidity pool sponsorship
+    #[arg(long, group = "sponsorship_type")]
+    pub liquidity_pool_id: Option<String>,
+
+    /// Claimable balance ID for claimable balance sponsorship. Accepts multiple formats:
+    /// - API format with type prefix (72 chars): 000000006f2179b31311fa8064760b48942c8e166702ba0b8fbe7358c4fd570421840461
+    /// - Direct hash format (64 chars): 6f2179b31311fa8064760b48942c8e166702ba0b8fbe7358c4fd570421840461
+    /// - StrKey format (base32): BAAMLBZI42AD52HKGIZOU7WFVZM6BPEJCLPL44QU2AT6TY3P57I5QDNYIA
+    #[arg(long, group = "sponsorship_type")]
+    pub claimable_balance_id: Option<String>,
+
+    /// Signer key for signer sponsorship
+    #[arg(long, group = "sponsorship_type")]
+    pub signer_key: Option<xdr::SignerKey>,
```

Comment:
I think the signer key should be a a string. All signer keys have strkey representations. And in theory someone will want to use aliases for stellar-cli managed keys as signers too, so we need alias lookups.

Reply:
Oh, yeah. I was supposed to get back to that and forgot. I even have a ticket to convert another command (#2214).